dart/dynamics/Joint.h

```diff
@@ -557,8 +620,19 @@ class Joint : public virtual common::Subject
   friend class Skeleton;
 
 protected:
+
+  /// Constructor called by inheriting class
+  Joint(const Properties& _properties);
+
+  /// Create a clone of this Joint. This may only be called by the Skeleton
+  /// class.
+  virtual Joint* clone() const = 0;
+
+  /// Called by the Skeleton class
+  virtual void registerDofs() = 0;
+
   /// Initialize this joint. This function is called by BodyNode::init()
-  virtual void init(Skeleton* _skel);
+  virtual void init(std::shared_ptr<Skeleton> _skel);
```

Comment:
I kept it in just in case someone ever makes a Joint type that requires initialization for some reason. But maybe that's an extremely unlikely use case.

Since Joint::init is completely internal (protected), we can just remove it instead of deprecating it.